src/main/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImpl.java

```diff
@@ -43,9 +44,19 @@
     @Override
     public CompanyData createCompanyData(final CompanySpec spec) throws DataException {
         if (spec == null) {
-            new IllegalArgumentException("CompanySpec can not be null");
+            throw new IllegalArgumentException("CompanySpec can not be null");
         }
-        String companyNumber = String.valueOf(randomService.getNumber(COMPANY_NUMBER_LENGTH));
+        String companyNumber;
+
+        switch (spec.getJurisdiction()){
+            case ENGLAND_WALES: companyNumber = String.valueOf(randomService.getNumber(COMPANY_NUMBER_LENGTH));
+            break;
+            case SCOTLAND: companyNumber = SCOTTISH_COMPANY_PREFIX +
+                    randomService.getNumber(COMPANY_NUMBER_LENGTH - SCOTTISH_COMPANY_PREFIX.length());
+            break;
+            default: companyNumber = String.valueOf(randomService.getNumber(COMPANY_NUMBER_LENGTH));
```

Comment:
I would put `case ENGLAND_WALES` and `default` one after the other so to use the same statement and avoid duplication. Although we'll eventually add NI and maybe we should throw an exception for the default case then